internal/rbd/replicationcontrollerserver.go

```diff
@@ -887,6 +880,29 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 	return resp, nil
 }
 
+func resyncVolume(localStatus librbd.SiteMirrorImageStatus, rbdVol *rbdVolume, force bool) error {
+	if resyncRequired(localStatus) {
+		// If the force option is not set return the error message to retry
+		// with Force option.
+		if !force {
+			return status.Errorf(codes.FailedPrecondition,
+				"image is in %q state, description (%s). Force resync to recover volume",
+				localStatus.State, localStatus.Description)
+		}
+		err := rbdVol.resyncImage()
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+
+		// If we issued a resync, return a non-final error as image needs to be recreated
+		// locally. Caller retries till RBD syncs an initial version of the image to
+		// report its status in the resync request.
+		return status.Error(codes.Unavailable, "awaiting initial resync due to split brain")
```

Comment:
`UNAVAILABLE` does not seem to be an error listed in [spec](https://github.com/csi-addons/spec/tree/main/replication#error-scheme-4)? Should this be `ABORTED` instead?

Reply:
Not part of this PR, it was added earlier

Reply:
As discussed will update the spec for the new error code

Reply:
Thanks, it seems this is intentional https://github.com/ceph/ceph-csi/pull/2507#pullrequestreview-754755312

Reply:
Yes correct.